Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/viewholder/SwitchSettingViewHolder.kt

```diff
@@ -57,6 +59,13 @@ class SwitchSettingViewHolder(
                 binding.settingSwitch.isEnabled = false
             }
 
+            if (setting.setting === BooleanSetting.MAIN_EMULATE_WII_SPEAK && isChecked) {
+                if (!PermissionsHandler.hasRecordAudioPermission(itemView.context)) {
+                  val settingsActivity = itemView.context as Activity
```

Comment:
This cast feels a bit fragile. How about: `adapter.fragmentView.fragmentActivity`